subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/repositories_cheat_sheet.adoc

```diff
@@ -0,0 +1,425 @@
+
+Consider the following GCS URL:
+
+----
+`gcs://myBucket/maven/release`
+----
+
+* `myBucket` is the GCS bucket name.
+* `/maven/release` is the GCS key (unique identifier for an object within a bucket)
+
+[[sub:maven_local]]
+== Declare a Local Maven Repository
+
+Gradle can consume dependencies from a link:https://maven.apache.org/guides/introduction/introduction-to-repositories.html[local Maven repository].
+Some teams use local Maven repositories to share artifacts between local projects.
+For instance, you might publish a library to the local repository and consume the library artifact from an application project.
+Don't use your local Maven repository as a cache: Gradle already stores resolved dependencies in <<dependency_resolution.adoc#sec:dependency_cache,its own cache>>.
+
+To add a local Maven repository to your build, add the following to your build script:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=maven-local]"]
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=maven-local]"]
+====
+
+Gradle uses the same logic as Maven to identify the location of your local Maven cache:
+
+* If your `settings.xml` defines a local repository, Gradle uses that location.
+* The `settings.xml` in `__USER_HOME__/.m2` takes precedence over the `settings.xml` in `__M2_HOME__/conf`.
+* If no `settings.xml` is available, Gradle uses the default location `__USER_HOME__/.m2/repository`.
+
+[[sec:case-for-maven-local]]
+=== Local Maven Repository Drawbacks
+
+Usually, you should not add a local Maven repository as a repository.
+Local Maven repositories impose the following drawbacks on your build:
+
+* The local Maven repository can contain partial dependencies, since Maven uses it as a cache. For example, if Maven never downloaded the source or javadoc files for a given module, Gradle will not find them either since it <<dependency_resolution.adoc#sec:how-gradle-downloads-deps,searches for files in a single repository>> once a module has been found. _This can prevent Gradle from building successfully._
+* Artifact origins cannot be tracked (a correctness and security problem).
+* Artifacts can be overwritten, _so Gradle does not perform any caching_.
+
+In some cases, you might have to use a local Maven repository:
+
+* Maven interoperability. For example, project A is built with Maven, project B is built with Gradle, and you need to share the artifacts during development.
+* Interoperability with Gradle itself. For example, you want to check that changes to project A work with project B. Usually you should use a <<composite_builds.adoc#composite_builds, composite build>> for this use case, but in some situations that isn't possible.
+
+If you do use a local Maven repository, combine it with <<declaring_repositories.adoc#sec:repository-content-filtering,a repository filter>> to ensure it only provides specific dependencies.
```

Comment:
nitpick: perhaps need another word, if you are forced to use, need to use, have to use, required to use... or just use
```suggestion
If you use a local Maven repository, combine it with <<declaring_repositories.adoc#sec:repository-content-filtering,a repository filter>> to ensure it only provides specific dependencies.
```

Reply:
In this case "do" is performing some heavy lifting -- it's emphasizing that this is a rare edge case that we don't recommend, but if you exist in a niche where you have no choice, you ought to do this. I'll add emphasis to make this clearer.

Reply:
```suggestion
If you _do_ use a local Maven repository, combine it with <<declaring_repositories.adoc#sec:repository-content-filtering,a repository filter>> to ensure it only provides specific dependencies.
```